src/car/car.service.ts

```diff
@@ -18,16 +18,16 @@ export class CarService {
         });
     }
 
-    async getCarUsage(carId: number): Promise<object> {
+    async getCarUsage(carId: number | object): Promise<object> {
         const monthName = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
 
         const query = `SELECT * FROM reservations WHERE car_id = ${carId}`;
 
-        const carData = await this.databaseService.executeQuery(query);
+        const carData = typeof carId === 'object' ? carId : await this.databaseService.executeQuery(query);
```

Comment:
what kind of object? and why do you use it instead of record from database?

Reply:
I was re-using this part for the "getAllCarsUsage" then, as per the word we had, it was decided to process all cars data as a whole. 
I forgot to replace the line 26 with the old one.